homeassistant/components/lutron/__init__.py

```diff
@@ -68,18 +68,19 @@ def setup(hass, base_config):
                 hass.data[LUTRON_DEVICES]["switch"].append((area.name, output))
         for keypad in area.keypads:
             for button in keypad.buttons:
-                # This is the best way to determine if a button does anything
-                # useful until pylutron is updated to provide information on
-                # which buttons actually control scenes.
-                for led in keypad.leds:
-                    if (
-                        led.number == button.number
-                        and button.name != "Unknown Button"
-                        and button.button_type in ("SingleAction", "Toggle")
-                    ):
-                        hass.data[LUTRON_DEVICES]["scene"].append(
-                            (area.name, keypad.name, button, led)
-                        )
+                # If the button has a function assigned to it, add it as a scene
+                if button.name != "Unknown Button" and button.button_type in (
+                    "SingleAction",
+                    "Toggle",
+                ):
+                    # Associate an LED with a button if there is one
+                    led = next(
+                        (led for led in keypad.leds if led.number == button.number),
+                        None,
+                    )
+                    hass.data[LUTRON_DEVICES]["scene"].append(
+                        (area.name, keypad.name, button, led)
```

Comment:
What will happen in the scene when the led is `None`?

Reply:
Nothing (that I'm aware of, LED isn't required from what I can tell) - it'll just add the scene. It's still executable from my testing.

Reply:
The led instance doesn't seem used at all in the home assistant scene instance. Why do we pass it to the scene?

Reply:
Now that I think of it, that is a really good question. Maybe the thought for it was that we could create a sensor that could keep track if a scene was active or not? Just a thought.... @cdheiser probably has a better idea than myself.

Reply:
@MartinHjelmare does it make sense to split the removal of LEDs out into a separate PR in case we introduce a breaking change? This way we can separate out fixes and get this fix put into production?